File: lib/payload_codec/payload_codec.ex
defmodule PayloadCodec do
  @external_resource "docs/payload_codec.md"
  @moduledoc File.read!("docs/payload_codec.md")

  def start_default do
    TemporalSdk.start_workflow(:cluster_1, "default", HelloWorld.Workflow, [
      :wait,
      input: [["from Temporal"]],
      workflow_id: "plain"
    ])
  end

  def start_encrypted do
    TemporalSdk.start_workflow(:cluster_1_enc, "encrypted", HelloWorld.Workflow, [
      :wait,
      input: [["from Temporal"]],
      workflow_id: "encrypted"
    ])
  end
end

File: src/payload_codec/payload_codec.erl
-module(payload_codec).

% elp:ignore W0012 W0040 E1599
-moduledoc {file, "../../docs/payload_codec.md"}.

-export([
    start_default/0,
    start_encrypted/0
]).

start_default() ->
    temporal_sdk:start_workflow(cluster_1, "default", hello_world_workflow, [
        wait, {input, [[~b"from Temporal"]]}, {workflow_id, "plain"}
    ]).

start_encrypted() ->
    temporal_sdk:start_workflow(cluster_1_enc, "encrypted", hello_world_workflow, [
        wait, {input, [[~b"from Temporal"]]}, {workflow_id, "encrypted"}
    ]).
